sdk/confidentialledger/test-resources.json

```diff
@@ -0,0 +1,109 @@
+{
```

Comment:
Preferably we start using bicep templates instead of ARM templates, if you don't mind updating. The file can be called `test-resources.bicep` and can look like below. Also we don't need to include parameters or outputs for the service principal/subscription values as the deployment script will handle that and prefix them in the way you had specified (similarly, you have an oid hardcoded default that needs to be removed). Additionally, the references to `CONFIDENTIALLEDGER_GROUP` should be changed to `CONFIDENTIALLEDGER_RESOURCE_GROUP` to follow convention (it gets set as an env var output [here](https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/TestResources/New-TestResources.ps1#L251)).

```
@description('Ledger Name')
@minLength(3)
@maxLength(24)
param ledgerName string = resourceGroup().name

@description('The client OID to grant access to test resources.')
param testApplicationOid string

@description('The location of the resource. By default, this is the same as the resource group.')
param location string = resourceGroup().location

var azureConfidentialLedgerUrl = 'https://${ledgerName}.confidential-ledger.azure.com'

resource ledgerName_resource 'Microsoft.ConfidentialLedger/ledgers@2022-05-13' = {
  name: ledgerName
  location: location
  properties: {
    ledgerType: 'Public'
    aadBasedSecurityPrincipals: [
      {
        principalId: testApplicationOid
        ledgerRoleName: 'Administrator'
      }
    ]
  }
}

output CONFIDENTIALLEDGER_ENDPOINT string = azureConfidentialLedgerUrl
```

Reply:
No prob. FYI bicep has a nifty `az bicep decompile -f <arm template>` function that makes it pretty easy to migrate.